main/app.c

```diff
@@ -2227,6 +2349,111 @@ int ast_app_group_set_channel(struct ast_channel *chan, const char *data)
 	return res;
 }
 
+int ast_app_group_set_var(struct ast_channel *chan, const char *group, const char *category, const char *name, const char *value)
+{
+	struct ast_group_meta *gmi = NULL;
+
+	struct varshead *headp;
+	struct ast_var_t *newvariable = NULL;
+
+	if (!group || !name) {
+		ast_log(LOG_WARNING, "<%s> GROUP assignment failed for %s@%s, group/name cannot be NULL, group variable '%s' not set\n", ast_channel_name(chan), group, category, name);
+		return -2;
+	}
+
+	if (!category) {
+		category = "";
+	}
+
+	if (!value) {
+		value = "";
+	}
+
+	/* Find our group meta data */
+	AST_RWLIST_WRLOCK(&groups_meta);
+	AST_RWLIST_TRAVERSE_SAFE_BEGIN(&groups_meta, gmi, group_meta_list) {
+		if ((strcasecmp(gmi->group, group) != 0) || (strcasecmp(gmi->category, category) != 0)) {
+			continue;
+		}
+
+		headp = &gmi->varshead;
+
+		AST_LIST_TRAVERSE(headp, newvariable, entries) {
+			if (strcasecmp(ast_var_name(newvariable), name) == 0) {
+				/* there is already such a variable, delete it */
+				AST_LIST_REMOVE(headp, newvariable, entries);
+				ast_var_delete(newvariable);
+				break;
+			}
+		}
+
+		newvariable = ast_var_assign(name, value);
+
+		AST_LIST_INSERT_HEAD(headp, newvariable, entries);
+		manager_event(EVENT_FLAG_DIALPLAN, "GroupVarSet",
+			"Channel: %s\r\n"
+			"Category: %s\r\n"
+			"Group: %s\r\n"
+			"Variable: %s\r\n"
+			"Value: %s\r\n"
+			"Uniqueid: %s\r\n",
+			chan ? ast_channel_name(chan) : "none",
```

Comment:
This could be simplified using S_OR, same with two lines below

Reply:
I think this is just syntactic sugar here.  There's no simplification between chan ? ... and S_SOR(x,y,z)